util/src/test/java/com/pkb/pkbcommon/ageutil/AgeUtilServiceTest.java

```diff
@@ -0,0 +1,43 @@
+package com.pkb.pkbcommon.ageutil;
+
+import com.pkb.common.datetime.DateTimeService;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import java.time.Instant;
+import java.time.LocalDate;
+import java.time.temporal.ChronoUnit;
+
+import static com.github.karsaig.approvalcrest.jupiter.MatcherAssert.assertThat;
+import static java.time.ZoneOffset.UTC;
+import static org.hamcrest.Matchers.is;
+import static org.mockito.Mockito.when;
+
+@ExtendWith(MockitoExtension.class)
+public class AgeUtilServiceTest {
+
+    @Mock
+    DateTimeService dateTimeService;
+    @InjectMocks
+    AgeUtilService underTest;
+
+    @Test
+    public void testUnderSixteen() {
+        Instant currentTime = Instant.parse("2020-01-02T00:00:00Z");
+        when(dateTimeService.now()).thenReturn(currentTime);
+        LocalDate dateOfBirth = currentTime.atZone(UTC).minus(16, ChronoUnit.YEARS).toLocalDate();
+        assertThat(underTest.patientIsOlderThan(dateOfBirth, 16), is(false));
+    }
+
+    @Test
+    public void testOverSixteen() {
+        Instant currentTime = Instant.parse("2020-01-02T00:00:00Z");
+        when(dateTimeService.now()).thenReturn(currentTime);
+        LocalDate dateOfBirth = currentTime.atZone(UTC).minus(16, ChronoUnit.YEARS).minus(1, ChronoUnit.DAYS).toLocalDate();
+        assertThat(underTest.patientIsOlderThan(dateOfBirth, 16), is(true));
+    }
+
```

Comment:
NB: Would be good to test with non-UTC timestamps given that the timezone is set on our deployments via the jvm user.timezone argument.

Reply:
The code which is under test doesn't actually rely on any `TimeZone` information, it's just comparing an `Instant` with a `LocalDate`. I introduced `TimeZone` into the test, so I could manipulate the date using the `minus` method, which actually complicated it and made it confusing. I've re-written it now, so please see what you think. thanks